lib/resourcemerge/imagestream_test.go

```diff
@@ -0,0 +1,818 @@
+package resourcemerge
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	imagev1 "github.com/openshift/api/image/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/utils/pointer"
+)
+
+func TestEnsureImageStreamv1(t *testing.T) {
+	tests := []struct {
+		name     string
+		existing imagev1.ImageStream
+		required imagev1.ImageStream
+
+		expectedModified bool
+		expected         imagev1.ImageStream
+	}{
+		{
+			name: "same lookup policy",
+			existing: imagev1.ImageStream{
+				Spec: imagev1.ImageStreamSpec{
```

Comment:
nit: looks like we could reduce a lot of visual clutter if our existing/required/actual data was just `ImageStreamSpec` (it seems that we always work with just this field) and we would build the `ImageStream` instances dynamically:

```go
existing := imagev1.ImageStream{Spec: tc.existing}
```

Alternatively, we could have testcases that show that `Status` part is not reconciled.

Reply:
Both nitpicks seems like good suggestions! I am working on the changes :+1:

Reply:
It does bring additional logic to the testing but I think the removal of the visual clutter is worth it. And the developer can focus only on the `ImageStreamSpec` when testing CVO behaviour on `ImageStreamSpec`. And `ImageStreamStatus` respectively.

---

> Alternatively, we could have testcases that show that Status part is not reconciled.

Yes, this test should be present! I have broken up the previous function `TestEnsureImageStreamv1` into `TestEnsureImageStreamv1_ImageStreamStatus` and `TestEnsureImageStreamv1_ImageStreamSpec` while keeping your suggestion to reduce the visual clutter in mind.

Let me know what you think of it!